repository/src/main/java/org/alfresco/repo/security/authentication/identityservice/IdentityServiceFacadeFactoryBean.java

```diff
@@ -554,12 +560,20 @@ private Collection<URI> possibleMetadataURIs()
 
     static class JwtDecoderProvider
     {
-        private static final SignatureAlgorithm SIGNATURE_ALGORITHM = SignatureAlgorithm.RS256;
+        private static final SignatureAlgorithm DEFAULT_SIGNATURE_ALGORITHM = SignatureAlgorithm.RS256;
         private final IdentityServiceConfig config;
+        private final Set<SignatureAlgorithm> signatureAlgorithms;
 
         JwtDecoderProvider(IdentityServiceConfig config)
         {
             this.config = requireNonNull(config);
+            this.signatureAlgorithms = ofNullable(config.getSignatureAlgorithms())
+                .filter(algorithms -> !algorithms.isEmpty())

```

Comment:
Using predicates and method references this could have been something like `filter(not(Set::isEmpty))` which is slightly more similar to natural language & avoids the 'algorithms' repetition, and therefore is slightly easier to read

Reply:
Done.